flow_shear.f90

```diff
@@ -156,17 +158,20 @@ subroutine advance_parallel_flow_shear(gout)
 
       allocate (g0k(naky, nakx))
 
-      do ivmu = vmu_lo%llim_proc, vmu_lo%ulim_proc
-         do it = 1, ntubes
-            do iz = -nzgrid, nzgrid
-               call get_dchidy(iz, ivmu, phi(:, :, iz, it), apar(:, :, iz, it), g0k)
-
-               !parallel flow shear
-               gout(:, :, iz, it, ivmu) = gout(:, :, iz, it, ivmu) + prl_shear(ia, iz, ivmu) * g0k
+      if (full_flux_surface) then
+         if (proc0) write (*, *) '!!!WARNING: flow shear not currently supported for full_flux_surface=T!!!'
+      else
```

Comment:
Should this error?

Also, I quite like to write preconditions such as this so that the main function code isn't in the else block:

```suggestion
      if (full_flux_surface) then
         if (proc0) write (*, *) '!!!WARNING: flow shear not currently supported for full_flux_surface=T!!!'
         call mp_abort(...)
         ! or just
         return
      end if
```

This way the rest of the function doesn't need to be indented. It's not a _massive_ benefit, but I do find it useful overall, especially if there are multiple preconditions.